android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -783,50 +783,59 @@ public VideoStreamingParameters getLastCachedStreamingParameters() {
-            Integer constraintHeightMin = range.getMinResolution().getResolutionHeight();
-            if (allCapabilities != null && !allCapabilities.isEmpty()) {
-
-                for (VideoStreamingCapability capability : allCapabilities) {
-                    double diagonal;
-                    if (capability.getPreferredResolution() == null
-                            || capability.getPreferredResolution().getResolutionHeight() == null
-                            || capability.getPreferredResolution().getResolutionWidth() == null) {
-                        continue;
-                    }
-                    if (capability.getDiagonalScreenSize() == null) {
-                        diagonal = parameters.getPreferredDiagonal();
-                    } else {
-                        diagonal = capability.getDiagonalScreenSize();
-                    }
-
-                    if (range.getMinScreenDiagonal() > diagonal) {
-                        continue;
-                    }
-
-                    if (!isAspectRatioInRange(range.getMinAspectRatio(), range.getMaxAspectRatio(), capability.getPreferredResolution())) {
-                        if (constraintHeightMax == null && constraintHeightMin == null) {
-                            continue;
-                        }
-                    }
+        if (this.supportedLandscapeStreamingRange == null && this.supportedPortraitStreamingRange == null){
+            for (VideoStreamingCapability capability: allCapabilities) {
+                capability.setAdditionalVideoStreamingCapabilities(null);
+            }
 
-                    if (!isImageResolutionInRange(range.getMinResolution(), range.getMaxResolution(), capability.getPreferredResolution())) {
-                        continue;
-                    }
+            validCapabilities.addAll(allCapabilities);
+        } else if (supportedLandscapeStreamingRange != null && supportedPortraitStreamingRange == null) {
+            for (VideoStreamingCapability capability : allCapabilities) {
+                if (determineResolutionType(capability.getPreferredResolution()) == ImageResolutionKind.LANDSCAPE){
+                    capability.setAdditionalVideoStreamingCapabilities(null);
+                    validCapabilities.add(capability);
+                }
+            }
+        } else if (supportedLandscapeStreamingRange == null && supportedPortraitStreamingRange != null) {
+            for (VideoStreamingCapability capability : allCapabilities) {
+                if (determineResolutionType(capability.getPreferredResolution()) == ImageResolutionKind.PORTRAIT) {
+                    capability.setAdditionalVideoStreamingCapabilities(null);
+                    validCapabilities.add(capability);
+                }
+            }
+        } else if (isZeroRange(supportedPortraitStreamingRange) && isZeroRange(supportedLandscapeStreamingRange)) {
+            return null;
+        }else {
```

Comment:
@RHenigan Hi! I am a little bit confused with this comment because from my understanding the null-safety is covered by the lines 803-817. Please, let me know If I am missing something - I will be ready to introduce the fix ASAP

<img width="905" alt="изображение" src="https://user-images.githubusercontent.com/20925367/109155354-1eb6a800-7778-11eb-8981-ebd87a3adf5f.png">

Reply:
Helo @kostyaBoss,
lines 803-817 here are checking if only one of the ranges is null, If one range is null then capabilities of that type are added to valid capabilities. But after that we still need to check against the non-null range and compare the capabilities to that range.

i.e. 
landScapeRange == null and portraitRange != null
All the landScape Capabilities are added to validCapabilities
Now we need to compare the capabilities received to the portraitRange that the developer has defined
Add any capabilities that match the portraitRange to validCapabilities

And the reverse would also need to happen if landScapeRange != null and portraitRange == null

With that being the case, I think this method can be simplified, looking at the iOS implementation [here](https://github.com/smartdevicelink/sdl_ios/blob/cb3d5b15a971f46b66e9738fc06b4bc3f98aac8b/SmartDeviceLink/private/SDLStreamingVideoLifecycleManager.m#L852)
getSupportedCapability would iterate over the list of capabilities
Identify the ImageResoultionType
call inRange with the capability and the correct Range
and inRange will return true if the range is null or the range matches, and false if the range does not match or is ZeroRange

Reply:
@RHenigan thanks for the detailed reply, checking now

Reply:
@RHenigan could you please confirm the changes are suitable? Pushed the latest version into this PR